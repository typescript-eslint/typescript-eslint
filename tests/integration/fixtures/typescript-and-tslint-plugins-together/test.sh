#!/bin/bash

# Generate the package.json to use
node /usr/utils/generate-package-json.js

# Install dependencies
npm install

# Use the local volumes for our own packages
npm install $(npm pack /usr/typescript-estree | tail -1)
npm install $(npm pack /usr/parser | tail -1)
npm install $(npm pack /usr/experimental-utils | tail -1)
npm install $(npm pack /usr/eslint-plugin-tslint | tail -1)
npm install $(npm pack /usr/eslint-plugin | tail -1)

# Run the linting
# (the "|| true" helps make sure that we run our tests on failed linting runs as well)
npx eslint --format json --output-file /usr/lint-output.json --config /usr/linked/.eslintrc.js /usr/linked/**/*.ts || true

# Run our assertions against the linting output
npx jest /usr/test.js --snapshotResolver=/usr/utils/jest-snapshot-resolver.js
