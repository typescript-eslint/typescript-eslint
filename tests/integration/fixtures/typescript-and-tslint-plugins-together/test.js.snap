// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should produce the expected lint ouput 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "/usr/linked/index.ts",
    "fixableErrorCount": 1,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 7,
        "endColumn": 13,
        "endLine": 1,
        "line": 1,
        "message": "'noSemi' is assigned a value but never used.",
        "messageId": "unusedVar",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 1,
      },
      Object {
        "column": 20,
        "endColumn": 20,
        "endLine": 1,
        "fix": Object {
          "range": Array [
            19,
            19,
          ],
          "text": ";",
        },
        "line": 1,
        "message": "Missing semicolon (tslint:semicolon)",
        "messageId": "failure",
        "nodeType": null,
        "ruleId": "@typescript-eslint/tslint/config",
        "severity": 2,
      },
    ],
    "source": "const noSemi = true
",
    "usedDeprecatedRules": Array [],
    "warningCount": 1,
  },
]
`;
